src/lib/discord-helpers.ts

```diff
@@ -1,51 +1,58 @@
-        });
+export const resetMessageReaction = async (messageReaction: MessageReaction): Promise<void> => {
+    await factory.defaultRetry<Message>(() => messageReaction.message.reactions.removeAll());
 };
 
-export const applyReactionToMessage = async function(messageReaction: MessageReaction, emojiName: string): Promise <void> {
-    await messageReaction.message.react(emojiName);
+export const applyReactionToMessage = async (messageReaction: MessageReaction, emojiName: string): Promise<void> => {
+    await factory.defaultRetry<MessageReaction>(() => messageReaction.message.react(emojiName));
 };
 
-export const extractUserCidFromGuildMember = function(member: GuildMember): string {
+export const extractUserCidFromGuildMember = (member: GuildMember): string => {
     const nickname = member && (member.nickname || member.user?.username);
-        if (!nickname) {
-            throw new AppError(`User's VATSIM CID could not be determined.`);
-        }
-        
-        let vatsimCid = nickname.substring(nickname.length - 7);
-        if (vatsimCid.length !== 7 || !+vatsimCid) {
-            throw new AppError(`User's VATSIM CID could not be determined.`);
-        }
+    if (!nickname) {
+        throw new AppError(`User's VATSIM CID could not be determined.`);
+    }
 
-        return vatsimCid;
+    let vatsimCid = nickname.substring(nickname.length - 7);
+    if (vatsimCid.length !== 7 || !+vatsimCid) {
+        throw new AppError(`User's VATSIM CID could not be determined.`);
+    }
+
+    return vatsimCid;
 };
 
-export const getVatsimUser = async function(vatsimApi: VatsimApi, cid: string): Promise<any> {
-    const apiInstance = await vatsimApi.getApiInstance();
-    return vatsimApi.getVatsimUser(apiInstance, cid);
+export const getVatsimUser = async (vatsimApi: VatsimApi, cid: string): Promise<any> => {
+    return factory.defaultRetry<any>(async () => {
+        const apiInstance = await vatsimApi.getApiInstance();
+        return vatsimApi.getVatsimUser(apiInstance, cid);
+    });
 };
 
-export const constructEmbeddedMessage = function(header: string, message: string, avatarUrl: string): MessageEmbed {
+export const constructEmbeddedMessage = (header: string, message: string, avatarUrl: string): MessageEmbed => {
     return new MessageEmbed()
         .setAuthor(header, avatarUrl)
         .setDescription(message);
 };
 
-export const sendMessageToUser = async function(header: string, message: string, user: User | PartialUser): Promise<void> {
-    // Should we be sending the bot's avatar instead?
-    const embeddedMessage = constructEmbeddedMessage(header, message, user.avatarURL());
-    await user.send(embeddedMessage);
+export const sendMessageToUser = async (header: string, message: string, user: User | PartialUser): Promise<void> => {
+    await factory.defaultRetry<Message>(() => {
+        // Should we be sending the bot's avatar instead?
```

Comment:
agreed, lets send the bot's avatar when talking to the user in DMs, but running the check command should return the user's avatar

Reply:
From all my testing, this only ever ends up using the bot's avatar .. Not really sure what changing this url does, but this is code was pre-existing my arrival ... We can add an issue to investigate

Reply:
If you look at a picture of the embedded section, there is an avatar at the top left. The avatar isn't needed but can sometimes having the avatar makes the styling nicer, maybe for sending DMs we don't need an avatar at all, but just the message. Keeping the avatar on a `¬check` command I think is a must, it helps identify the user we are checking. What do you think about removing the avatar from direct message communications all together?

Reply:
Oh yes, I see what you mean.

When sending a message to a channel, this is what you see:

![image](https://user-images.githubusercontent.com/5215509/96870997-f76e6800-1471-11eb-84b5-cd8f602b90bf.png)

When the bot messages you direct, you get:

![image](https://user-images.githubusercontent.com/5215509/96871058-08b77480-1472-11eb-85c7-318c10da9662.png)

Maybe embeddedMessages in messages direct to the user aren't the right option?

Reply:
Agreed, I will make this a separate issue as this probably isn't the best place to discuss it